Src/IronPython/Runtime/Operations/PythonOps.cs

```diff
@@ -2833,38 +2833,44 @@ public static partial class PythonOps {
         }
 
         public static object? ConvertFloatToComplex(object value) {
-            if (value == null) {
-                return null;
-            }
-
-            if (value is double d) return new Complex(d, 0.0);
-            if (value is Extensible<double> ed) return new Complex(ed.Value, 0.0);
-            throw new InvalidOperationException();
+            return value switch {
+                null => null,
+                double d => new Complex(d, 0.0),
+                Extensible<double> ed => new Complex(ed.Value, 0.0),
+                _ => throw new InvalidOperationException(),
+            };
         }
 
         public static object? ConvertIntToBigInt(object? value) {
             return value switch {
                 null => null,
                 int i => new BigInteger(i),
                 BigInteger bi => bi,
+                Extensible<BigInteger> ebi => ebi.Value,
                 _ => throw new InvalidOperationException(),
             };
         }
 
-        internal static bool CheckingConvertToInt(object value) {
-            return value is int || value is BigInteger || value is Extensible<BigInteger>;
+        public static object? ConvertIntToInt32(object? value) {
+            return value switch {
+                null => null,
+                int i => i,
```

Comment:
Probably insignificant optimization, but maybe return `value` here? I guess the same applies to `ConvertIntToBigInt`.

Reply:
I thought about it and took the lead from `ConvertToPythonPrimitive` (a few lines above), which does more of this kind of stuff, presumably for readability (?).

OK to optimize it.